src/cli/run.ts

```diff
@@ -0,0 +1,84 @@
+import * as github from "@actions/github";
+import {getHistoryProvider} from "../history/index.js";
+import {resolveShouldPersist} from "../history/shouldPersist.js";
+import {validateBenchmark} from "../history/schema.js";
+import {Benchmark, BenchmarkOpts, FileCollectionOptions, StorageOptions} from "../types.js";
+import {renderCompareWith, resolveCompareWith, resolvePrevBenchmark} from "../compare/index.js";
+import {parseBranchFromRef, getCurrentCommitInfo, shell, getCurrentBranch, collectFiles} from "../utils/index.js";
+import {computeBenchComparision} from "../compare/compute.js";
+import {postGaComment} from "../github/comment.js";
+import {isGaRun} from "../github/context.js";
+import {BenchmarkRunner} from "../benchmark/runner.js";
+import {optionsDefault} from "./options.js";
+import {consoleLog} from "../utils/output.js";
+
+export async function run(opts_: FileCollectionOptions & StorageOptions & BenchmarkOpts): Promise<void> {
+  const opts = Object.assign({}, optionsDefault, opts_);
+
+  // Sanitize opts
+  if (isNaN(opts.threshold)) throw Error("opts.threshold is not a number");
```

Comment:
shouldnt yargs do this sanitization?

Reply:
Should this option also get checked `coerceToOptsObj` for being set incorrectly at the benchIt level?